builtin/providers/ns1/metadata.go

```diff
@@ -0,0 +1,535 @@
+package ns1
+
+import (
+	"fmt"
+	"log"
+	"math"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"gopkg.in/ns1/ns1-go.v2/rest/model/data"
+)
+
+func metadataSchema() *schema.Schema {
```

Comment:
The main issue Ive been stuggling with is modeling the ns1 metadata structure. NS1 metadata can be attached at three different levels of a dns record, and is dynamic/flexible in nature.

From the [ns1 api docs](https://ns1.com/api/#record-put):

> Metadata tables (meta) may be specified in answers, in regions or in the record as a whole. (Note that regions can be any kind of group, not just geo-related. For this reason, the NS1 web portal at my.nsone.net uses the word "group" instead of "region".) The metadata tables may contain key/value pairs where valid keys and values are as described in the output of /metatypes.

> Anywhere metadata tables can go, data feeds can go as well. Connect a compatible feed to a metadata key by assigning its value to be a JSON object pointing to the data feed that should be consulted for the final value.

Metadata is also modeled in the [ns1 go rest client](https://github.com/ns1/ns1-go/blob/v2/rest/model/data/meta.go)

 
Not sure if `TypeList` of `MaxItems=1` is the best way to construct ns1 metadata. The main issues ive run into are:

 - "static fields" and "dynamic fields" should be in conflict with one another. But because im trying to make `metadataSchema` a reusable/shared struct for `record`s, `record.answer`s, and `record.region`s, im not sure how to use `ConflictsWith` for each "static" metadata field and its "dynamic" counterpart. 
 
 - With issue https://github.com/hashicorp/terraform/issues/5694 in the works, im struggling to get default bool/int/float values working. A good example is the boolean `metadata.Up` field, which has a different meaning when absent as opposed to `false`.
 
So I guess for ns1 metadata, does it look like im on the right track here? It feels like im not using the `helper` package correctly on this structure...